setup_undercloud.yml

```diff
@@ -59,63 +59,102 @@
-
-      when: osp_release|int > 14
 
+        - name: set undercloud to PXE boot (Supermicro)
+          shell: ipmitool -I lanplus -H mgmt-{{ undercloud_hostname }} -U quads -P {{ chassis_password }} chassis bootdev pxe options=persistent
+          when: lab_name == "scale" and undercloud_hostname.split('.')[0].split('-')[3] in scale.machine_types.supermicro
+
+        - name: power cycle undercloud (Supermicro)
+          shell: ipmitool -I lanplus -H mgmt-{{ undercloud_hostname }} -U quads -P {{ chassis_password }} chassis power cycle
+          when: lab_name == "scale" and undercloud_hostname.split('.')[0].split('-')[3] in scale.machine_types.supermicro
+
+        - name: set undercloud to PXE boot off Foreman (Dell)
+          shell: ./badfish.py -H mgmt-{{ undercloud_hostname }} -u quads -p {{ chassis_password }} -i config/idrac_interfaces.yml --boot-to-type foreman --pxe
+          args:
+            chdir: "{{ ansible_user_dir }}/badfish"
+          when: lab_name == "scale" and undercloud_hostname.split('.')[0].split('-')[3] in scale.machine_types.dell
+
+        - name: power cycle undercloud (Dell)
+          shell: ./badfish.py -H mgmt-{{ undercloud_hostname }}  -u quads -p {{ chassis_password }} --reboot-only
+          args:
+            chdir: "{{ ansible_user_dir }}/badfish"
+          when: lab_name == "scale" and undercloud_hostname.split('.')[0].split('-')[3] in scale.machine_types.dell
+
+        - name: wait for 420 seconds before checking for undercloud
+          wait_for:
+            timeout: 420
+
+        - name: waiting for the undercloud to be available
+          wait_for:
+            port: 22
+            host: "{{ hostvars['localhost']['undercloud_hostname'] }}"
+            search_regex: OpenSSH
+            timeout: 30
+          register: uc_reachable
+          delegate_to: localhost
+          retries: 100
+          until: uc_reachable|succeeded
+
+        - name: add keys to new undercloud
+          include_tasks: tasks/copykeys.yml
+          vars:
+            hostname: "{{ undercloud_hostname }}"
+      when: rhel7_install is not skipped or rhel8_install is not skipped
+
+    - name: add undercloud_host to inventory file (RHEL 8)
+      add_host:
+        name: "{{ undercloud_hostname }}"
+        groups: "baremetal,undercloud,tester"
+        ansible_host: "{{ undercloud_hostname }}"
+        ansible_user: "{{ ansible_ssh_user }}"
+        ansible_ssh_private_key_file: "{{ ansible_ssh_key }}"
+        ansible_python_interpreter: /usr/libexec/platform-python
+      when: rhel8_install is not skipped
+
+    - name: add undercloud_host to inventory file (RHEL 7)
+      add_host:
+        name: "{{ undercloud_hostname }}"
+        groups: "baremetal,undercloud,tester"
+        ansible_host: "{{ undercloud_hostname }}"
+        ansible_user: "{{ ansible_ssh_user }}"
+        ansible_ssh_private_key_file: "{{ ansible_ssh_key }}"
+      when: rhel8_install is not skipped
```

Comment:
typo, patch is good to go after fixing this typo.

Reply:
Good catch.

Reply:
@venkataanil fixed